FreeRTOS/FreeRTOS
# Folder structure change + Fix broken Projects

Description
-----------
This PR intends to create a folder structure which is easy to navigate and clean. While moving the folders around, it intends to separate the Source and the Tests of FreeRTOS-Plus-TCP.
Both changes mentioned above will break the TCP test project. Therefore, the fix for the broken project is also included in the same PR.

The folder structure which the PR intends to achieve:
```
FreeRTOS
FreeRTOS-Plus
        + Demo
        + Source
                + FreeRTOS-Plus-<Lib_name>
                + FreeRTOS-Plus-TCP -> Remove the ‘test’ folder and take it to <root>/Test/FreeRTOS-Plus-TCP/Integration/Full-TCP-Networkless
        + Test
                + CBMC
                + Cmock
                + FreeRTOS-Plus-TCP
                        + Integration
                                + Full-TCP-Networkless -> add the above mentioned ‘test’ folder here
                        + Unit
```


There are 3 commits in the PR:
- First one cleans up the folders and creates new ones
- Second one fixes the Projects which broke due to folder structure change
- Third one moves the test folder to the ideal location

Test Steps
-----------
- Clone the repository `git clone https://github.com/AniruddhaKanhere/FreeRTOS.git --recurse-submodules ./Folder`
- Checkout the branch: `git checkout Folder_structure_change`
- Go to `FreeRTOS-Plus\Test\FreeRTOS-Plus-TCP\Integration\Full-TCP-Networkless` and open `Full-TCP-Networkless.sln`.
- Edit the configNETWORK_INTERFACE_TO_USE in FreeRTOS_Config.h.
- Compile and run! 4 tests should run...



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
